tests/returntypes/test_jax_new.py

```diff
@@ -32,9 +32,38 @@
 from pennylane.interfaces.jax_jit import _execute_with_fwd
 
 
-# TODO: add jax-jit when it supports new return types.
-# "jax-jit"
-@pytest.mark.parametrize("interface", ["jax-python"])
+@pytest.mark.parametrize(
+    "version, package, should_raise",
+    [
+        ("0.3.16", jax, True),
+        ("0.3.17", jax, False),
+        ("0.3.18", jax, False),
+        ("0.3.14", jax.lib, True),
+        ("0.3.15", jax.lib, False),
+        ("0.3.16", jax.lib, False),
+    ],
+)
+def test_raise_version_error(package, version, should_raise, monkeypatch):
+    """Test JAX version error"""
+    a = jnp.array([0.1, 0.2])
+
+    dev = qml.device("default.qubit", wires=1)
+
+    with qml.tape.QuantumTape() as tape:
+        qml.expval(qml.PauliZ(0))
+
+    with monkeypatch.context() as m:
+        m.setattr(package, "__version__", version)
+
+        if should_raise:
+            msg = "requires version 0.3.17 or higher for JAX and 0.3.15 or higher JAX lib"
+            with pytest.raises(InterfaceUnsupportedError, match=msg):
+                execute([tape], dev, gradient_fn=param_shift, interface="jax-jit")
+        else:
+            execute([tape], dev, gradient_fn=param_shift, interface="jax-jit")
+
+
+@pytest.mark.parametrize("interface", ["jax-python", "jax-jit"])
```

Comment:
I think it would be great to separate the files between jitting and non jitting and treat jax-jit like a whole interface. We will have some code duplication but the tests will easier to write and we will be sure to use jitting everywhere.

Reply:
Done